src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
```

Comment:
It's a good practice to validate the length of `partsOfDate` after splitting to ensure we have the correct number of parts before proceeding. This is more about defensive programming and making sure the function behaves correctly even if future changes in the code introduce unexpected input formats.